cmd/podman/libpodruntime/runtime.go

```diff
@@ -1,17 +1,59 @@
 package libpodruntime
 
 import (
+	"fmt"
+	"os"
+	"path/filepath"
+
 	"github.com/containers/storage"
 	"github.com/projectatomic/libpod/libpod"
 	"github.com/urfave/cli"
 )
 
 // GetRuntime generates a new libpod runtime configured by command line options
 func GetRuntime(c *cli.Context) (*libpod.Runtime, error) {
-	storageOpts := storage.DefaultStoreOptions
+	storageOpts, err := GetDefaultStoreOptions()
+	if err != nil {
+		return nil, err
+	}
 	return GetRuntimeWithStorageOpts(c, &storageOpts)
 }
 
+func GetRootlessStorageOpts() (storage.StoreOptions, error) {
+	var opts storage.StoreOptions
+
+	home := os.Getenv("HOME")
+	runtimePath := os.Getenv("XDG_RUNTIME_DIR")
+	if runtimePath == "" {
+		runtimePath = filepath.Join(home, "rundir")
```

Comment:
This environment variable is [from the XDG spec][1], which has:

> If `$XDG_RUNTIME_DIR` is not set applications should fall back to a replacement directory with similar capabilities and print a warning message.

So we should probably log a warning here.  I also think falling back to something under `TMPDIR` is closer to "a replacement directory with similar capabilities" than falling back to `~/rundir`.  Can we use `/tmp/libpod-{uid}` or some such as the fallback?

[1]: https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.7.html#variables

Reply:
let's revisit this later in a follow-up PR.  In any case it should be probably be fixed first in runc as it is uses only `XDG_RUNTIME_DIR`.  If it is not specified I don't think it is possible to run any container as rootless user.

Reply:
Why not add it to the environment before running runc, If we choose /tmp/libpod-user... or ~/runtime then set the XDG_RUNTIME_DIR to it and exec runc.


If we use /tmp, you should fail if the selected directory already exists and create it as 700 permissions, That should prevent the race condition that Miloslav talked about

Reply:
done, I've implemented this logic and now `XDG_RUNTIME_DIR` is set if it wasn't set and we created the directory somewhere else.